### DISCUSSION TITLE
Nous A1 no negative energy readings

### DISCUSSION THREAD

**Discussion by joba-1** [arendst/Tasmota#16255] [Q&A]
Hi,

today I used the Nous A1 for reading solar panel energy production and it seems to basically work.
Except that the values are not negative. If there is no sun there is some 1 W standby consumption and with sun also positive values.

Does anyone know if it is even possible with this device to distinguish power flow direction?

If there is nothing in the works, and no known show stopper, I'll probably have a go on this soon.
I've [seen something similar done to Shelly EM](https://github.com/arendst/Tasmota/discussions/12861), so I know where to look for a start...

**Comment by barbudor** [comment]
A plug is not intended to be used that way
Capability may depends on both the hardware design and the driver.
The Shelly EM, Sonoff Dual etc... are better designed for that tasl

**Reply by joba-1** [reply]
Thanks for the fast response.

> A plug is not intended to be used that way

probably not. But also true for most tasmota devices ;)

> Capability may depends on both the hardware design and the driver.

That's why I asked - I hope for someone that played around with it before me.

> The Shelly EM, Sonoff Dual etc... are better designed for that tasl

I have a plug and a socket - the A1 seems like a perfect fit physically. Would hate if I need to wire something extra and provide a housing.
Also the Sonoff Duals I have are not very reliable. They often get stuck - sometimes no more energy readings, sometimes completely gone. Cured just by power off/on.

So I'm not giving up, yet.

**Reply by barbudor** [reply]
As there seems to be different Nous A1 and A1T around, which template are you using ? (to known which chip and driver is used)

**Comment by joba-1** [comment]
Bought some time ago - A1T was only announced, not available
```
20:39:09.939 CMD: template
20:39:09.947 MQT: stat/nous1/RESULT = {"NAME":"NOUS A1","GPIO":[320,0,576,0,2656,2720,0,0,2624,32,0,224,0,0],"FLAG":0,"BASE":45}
```

**Reply by joba-1** [reply]
Not 100% sure, but I think it's a hlw8012. 
From its datasheet and example cirquit it looks like detecting power direction is not possible (it is chinese and I'm not, so maybe I get it wrong).
At least it also looks like measurement is symmetric, and me letting the power flow in reverse should not harm the device.

**Reply by sfromis** [reply]
According to the template for [Nous A1T](https://templates.blakadder.com/nous_A1T.html), it is same energy monitoring as [Nous A1](https://templates.blakadder.com/nous_A1.html), but different gpio layout.
